### PR TITLE
codecatalyst: Cloud9 fixes

### PR DESCRIPTION
**Pull request by nkomonen-amazon**
This fixes:
- CC auto connects on start to an existing valid connection
- CW Connection Prompter in C9 shows correct options when clicking `Start` node under CodeWHisperer Developer Tools
- The proceed to browser message contains the code since it was missing in C9 since C9 doesn't support the `detail` field in modal messages.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codewhisperer/util/showSsoPrompt.ts [hunk 1]
```diff
@@ -15,27 +15,14 @@ import { ToolkitError } from '../../shared/errors'
...
-        } else {
-            await AuthUtil.instance.connectToAwsBuilderId()
-        }
-
-        return
-    }
+export const showCodeWhispererConnectionPrompt = async () => {
+    const items = isCloud9('classic')
+        ? [createSsoItem(), createCodeWhispererIamItem()]
+        : [createBuilderIdItem(), createSsoItem(), createCodeWhispererIamItem()]
```
Comment (justinmk3) [inline src/codewhisperer/util/showSsoPrompt.ts:23]: Should show builder id for cloud9-codecatalyst? https://github.com/aws/aws-toolkit-vscode/pull/4058/commits/661549f62b0142fc5de525bdf977b2ed609579e3

File: src/codecatalyst/activation.ts [hunk 2]
```diff
@@ -35,6 +35,8 @@ export async function activate(ctx: ExtContext): Promise<void> {
     const commands = new CodeCatalystCommands(authProvider)
     const remoteSourceProvider = new CodeCatalystRemoteSourceProvider(commands, authProvider)
 
+    await authProvider.restore()
```
Comment (justinmk3) [inline src/codecatalyst/activation.ts:38]: which commit (accidentally) removed this? is likely worth reviewing it

Comment (nkomonen-amazon) [reply src/codecatalyst/activation.ts:38]: Yup. Are we sure that this was actually here before?

File: src/codewhisperer/util/showSsoPrompt.ts [hunk 3]
```diff
@@ -15,27 +15,14 @@ import { ToolkitError } from '../../shared/errors'
...
-            if (iamConn) {
-                await AuthUtil.instance.auth.useConnection(iamConn)
-            }
-        } else {
-            await AuthUtil.instance.connectToAwsBuilderId()
-        }
-
-        return
-    }
+export const showCodeWhispererConnectionPrompt = async () => {
```
[the remark this answers is not in the document]
Comment (nkomonen-amazon) [reply src/codewhisperer/util/showSsoPrompt.ts:20]: This is a CW specific connection prompter (eg builder id CW scopes are added) so I didn't want it to be generic

File: src/codewhisperer/util/showSsoPrompt.ts (hunk 1 above)
Comment (nkomonen-amazon) [reply src/codewhisperer/util/showSsoPrompt.ts:23]: This specifies `isCloud9('classic')` so it should show Builder ID for this when C9-codecatalyst. Am I misunderstanding `isCloud()`?

Comment (justinmk3) [reply src/codewhisperer/util/showSsoPrompt.ts:23]: here is what i was thinking: https://github.com/aws/aws-toolkit-vscode/commit/661549f62b0142fc5de525bdf977b2ed609579e3

Comment (justinmk3) [reply src/codewhisperer/util/showSsoPrompt.ts:23]: oh i missed that, the old code had `isCloud9()` but yeah this makes sense.

File: src/codecatalyst/activation.ts (hunk 2 above)
Comment (justinmk3) [reply src/codecatalyst/activation.ts:38]: no but it used to autoconnect

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**